### ISSUE TITLE
When coding in TypeScript, sibling sub-classes overwrite associations

### ISSUE THREAD

**Issue by jrrbru** [rjz/backbone.associate#15]
This issue arises when coding in TypeScript. I haven't looked to see if it arises also when using JavaScript.

If I create a Model that extends from Backbone.Model and add an association
```typescript
import OtherModel = require( './othermodel' ); // typescript import
var Backbone = require('backbone-associate'); // commonjs import

class SuperModel extends Backbone.Model { }
Backbone.associate(SuperModel , {
	other: { type: OtherModel }
});
```
and then create a subclass of `SuperModel` like so
```typescript
import AnotherModel = require( './anothermodel' ); // typescript import
var Backbone = require('backbone-associate'); // commonjs import

class SubModel extends SuperModel { }
Backbone.associate(SubModel, {
	another: { type: AnotherModel }
});
```
and finally I create another subclass of `SuperModel` that uses the same association, but to a different model, like this
```typescript
import CustomModel = require( './custommodel' ); // typescript import
var Backbone = require('backbone-associate'); // commonjs import

class SiblingModel extends SuperModel { }
Backbone.associate(SiblingModel, {
	another: { type: CustomModel}
});
```
The SiblingModel's association overwrites SubModel's association. My expected result is to see each sibling sub-model with their own specified association for "another".

**Comment by rjz** [comment]
@jrrbru, thanks for the report and diagnosis! I think we can set up a valid test case in vanilla JS along the lines of [this gist](https://gist.github.com/rjz/1b4cd721b71fc7d6ce18), and I'd be happy to merge a fix.

**Comment by rjz** [comment]
Closed in #16 (HT @jrrbru), and republished to npm / bower as 0.10.0.

**Comment by jrrbru** [comment]
Thanks for pulling it in.
